lib/facter/package_provider.rb

```diff
@@ -14,6 +14,11 @@
 
 # These will be nil if Puppet is not available.
 Facter.add(:package_provider) do
+
+  confine do
+    Facter.value(:kernel) != 'windows'
+  end
```

Comment:
You can even use weights. It still works on Windows so just give this a weight of 50:
```suggestion
  has_weight 50
```